source/agora/node/TransactionPool.d

```diff
@@ -790,6 +791,69 @@ unittest
     }
 }
 
+// test double-spend selection mechanism with maximum fee selector
+unittest
+{
+    import agora.consensus.Fee;
+    import agora.consensus.state.UTXOSet;
+
+    auto utxo_set = new TestUTXOSet();
+    auto fee_man = new FeeManager();
+    DoubleSpentSelector selector =
+                (Transaction[] txs)
+                {
+                    return maxIndex!((a, b) =>
+                        fee_man.getTXFee(a, &utxo_set.peekUTXO) <
+                        fee_man.getTXFee(b, &utxo_set.peekUTXO))(txs);
+                };
+    auto pool = new TransactionPool(new ManagedDatabase(":memory:"), selector);
+
+    // parent transaction
+    Transaction tx1 =
+    {
+        TxType.Payment,
+        [Input(Hash.init, 0)],
+        [Output(Amount(10), KeyPair.random.address)]
+    };
+
+    utxo_set.put(tx1);
+
+    // double spent transaction, transaction trying to spend parent
+    Transaction tx2 =
+    {
+        TxType.Payment,
+        [Input(tx1.hashFull(), 0)],
+        [Output(Amount(2), KeyPair.random.address)]
+    };
+
+    // double spent transaction, trying to spend parent
+    Transaction tx3 =
+    {
+        TxType.Payment,
+        [Input(tx1.hashFull(), 0)],
+        [Output(Amount(1), KeyPair.random.address)]
+    };
```

Comment:
Those transactions are not valid. Please use valid transactions in tests.

Reply:
if you mean 
[Input(Hash.init, 0)],
is not a valid input, then I see this pattarn in our code everywhere, I quick search reveals that every single Agora developer(except Chris) wrote tests like this :)

should we clean those up too?

Reply:
> is not a valid input, then I see this pattern in our code everywhere

And we suffer from it every week :(

Reply:
okay, let me fix this now, and also will also raise a ticket for cleaning up the entire codebase, and will do when I have more time